homeassistant/components/shelly/binary_sensor.py

```diff
@@ -223,9 +228,13 @@ def is_on(self) -> bool | None:
 class RpcBinarySensor(ShellyRpcAttributeEntity, BinarySensorEntity):
     """Represent a RPC binary sensor entity."""
 
+    entity_description: RpcBinarySensorDescription
+
     @property
-    def is_on(self) -> bool:
+    def is_on(self) -> bool | None:
         """Return true if RPC sensor state is on."""
+        if self.attribute_value is None:
+            return None
```

Comment:
This is a slight change in logic, as binary sensors can return `None` (see #60193)